tooling/templatize/pkg/pipeline/arm.go

```diff
@@ -207,10 +207,12 @@ func pollAndPrint[T any](ctx context.Context, p *runtime.Poller[T]) error {
 	return nil
 }
 
-func doDryRun(ctx context.Context, client *armresources.DeploymentsClient, rgName string, deploymentName string, step *types.ARMStep, pipelineWorkingDir string, cfg config.Configuration, input Outputs) (Output, error) {
+func doDryRun(ctx context.Context, client *armresources.DeploymentsClient, rgName string, deploymentName string, step *types.ARMStep, pipelineWorkingDir string, cfg config.Configuration, state *ExecutionState) (Output, error) {
 	logger := logr.FromContextOrDiscard(ctx)
 
-	inputValues, err := getInputValues(step.Variables, cfg, input)
+	state.RLock()
+	inputValues, err := getInputValues(step.Variables, cfg, state.Outputs)
+	state.RUnlock()
```

Comment:
prefer getters for things you need.  If they must match exactly, one function with two return values.